kof-operator/webapp/collector/tests/HealthSelector.test.tsx

```diff
@@ -0,0 +1,156 @@
+    expect(screen.getByText("0 Up")).toBeInTheDocument();
+    expect(screen.getByText("0 Unknown")).toBeInTheDocument();
+  });
+
+  it("should disable checkboxes when loading", async () => {
+    render(
+      <PrometheusTargetProvider>
+        <HealthSelector />
+      </PrometheusTargetProvider>
+    );
+
+    const checkboxes = screen.getAllByRole("checkbox") as HTMLInputElement[];
+
+    checkboxes.forEach((checkbox) => {
+      expect(checkbox.disabled).toBe(true);
+    });
+
+    await waitFor(() => {
+      checkboxes.forEach((checkbox) => {
+        expect(checkbox.disabled).toBe(false);
+      });
+    });
+  });
+
+  it("should display correct target counts for each health state", async () => {
+    render(
+      <PrometheusTargetProvider>
+        <HealthSelector />
+      </PrometheusTargetProvider>
+    );
+
+    const upCount = manager.targets.filter(
+      (target) => target.health === "up"
+    ).length;
+    const downCount = manager.targets.filter(
+      (target) => target.health === "down"
+    ).length;
+    const unknownCount = manager.targets.filter(
+      (target) => target.health === "unknown"
+    ).length;
+
+    await waitFor(() => {
+      const checkboxes = screen.getAllByRole("checkbox") as HTMLInputElement[];
+      checkboxes.forEach((checkbox) => {
+        expect(checkbox.disabled).toBe(false);
+      });
+
+      expect(screen.getByText(`${unknownCount} Unknown`)).toBeInTheDocument();
+      expect(screen.getByText(`${downCount} Down`)).toBeInTheDocument();
+      expect(screen.getByText(`${upCount} Up`)).toBeInTheDocument();
+    });
+  });
+
+  it("should handle single checkbox selection", async () => {
+    render(
+      <PrometheusTargetProvider>
+        <HealthSelector />
+        <TargetList />
+      </PrometheusTargetProvider>
+    );
+
+    await waitFor(() => {
+      const checkboxes = screen.getAllByRole("checkbox") as HTMLInputElement[];
+      checkboxes.forEach((checkbox) => {
+        expect(checkbox.disabled).toBe(false);
+      });
+    });
+
+    const upCount = manager.targets.filter(
+      (target) => target.health === "up"
+    ).length;
+
+    const checkboxes = screen.getAllByRole("checkbox");
+    const upCheckbox = checkboxes[2];
+
+    await userEvent.click(upCheckbox);
+    const filteredRows = screen.getAllByRole("row");
+
+    // 1 header + 2 target
+    expect(filteredRows).toHaveLength(upCount + 1);
+  });
+
+  it("should handle multiple checkbox selections", async () => {
+    render(
+      <PrometheusTargetProvider>
+        <HealthSelector />
+        <TargetList />
+      </PrometheusTargetProvider>
+    );
+
+    await waitFor(() => {
+      const checkboxes = screen.getAllByRole("checkbox") as HTMLInputElement[];
+      checkboxes.forEach((checkbox) => {
+        expect(checkbox.disabled).toBe(false);
+      });
+    });
+
+    const checkboxes = screen.getAllByRole("checkbox");
+    const upCheckbox = checkboxes[2];
```

Comment:
What is this magic constant 2 means? Probably better to define a separate role, like "up-checkbox" for better  readability.